eng/pipelines/templates/stages/source-build-and-validate.yml

```
@@ -89,8 +89,9 @@ stages:
           pool: ${{ parameters.pool_Linux }}
         buildSourceOnly: true
         ${{ if eq(parameters.useMicrosoftBuildAssetsForTests, false) }}:
-          extraProperties: "/p:DisableDevBuildAsDefaultForSourceOnly=true"
+          extraProperties: "${{ leg.officialBuildId }} /p:DisableDevBuildAsDefaultForSourceOnly=true"
```

We don't want to set `DisableDevBuildAsDefaultForSourceOnly` anymore. That will cause the build IDs from the git-info files to be used for each repo: https://github.com/dotnet/dotnet/blob/5d437e96e412b189af05abb84753996de9aa698f/repo-projects/Directory.Build.props#L20

```suggestion
          extraProperties: "${{ leg.officialBuildId }}"
```